Sources/PIRProcessDatabase/ProcessDatabase.swift

```diff
@@ -411,14 +386,78 @@ struct ProcessDatabase: ParsableCommand {
         }
     }
 
-    mutating func run() throws {
+    private func processShard<Scheme: HeScheme>(
+        shardID: String,
+        shard: KeywordDatabaseShard,
+        config: ResolvedArguments,
+        context: Context<Scheme>,
+        processArgs: ProcessKeywordDatabase.Arguments<Scheme>) async throws -> EvaluationKeyConfig
+    {
+        func logEvent(event: ProcessKeywordDatabase.ProcessShardEvent) throws {
+            let log: (String) -> Void = { message in ProcessDatabase.logger.info("Shard №\(shardID): \(message)") }
```

Comment:
I my opinion it would be better to use the [metadata api](https://github.com/apple/swift-log?tab=readme-ov-file#logging-metadata) to add the `shardID` to the logger.

```suggestion
            var logger = ProcessDatabase.logger
            logger[metadataKey: "shardID"] = shardID
```